openmls/src/group/mls_group/process.rs

```diff
@@ -33,13 +33,37 @@ impl MlsGroup {
         //  - ValSem6
         let decrypted_message = match message.wire_format() {
             WireFormat::MlsPlaintext => DecryptedMessage::from_inbound_plaintext(message)?,
-            WireFormat::MlsCiphertext => DecryptedMessage::from_inbound_ciphertext(
-                message,
-                self.ciphersuite(),
-                backend,
-                self.epoch_secrets(),
-                self.secret_tree_mut().deref_mut(),
-            )?,
+            WireFormat::MlsCiphertext => {
+                // If the message is older than the current epoch, we need to fetch the correct secret tree first
+                let ciphersuite = self.ciphersuite().clone();
+                let message_secrets = if message.epoch() < self.context().epoch() {
```

Comment:
So there are two different places with message secrets now. The one you get through `message_secrets_mut` and the store. This looks like a footgun.
How about we always use the `message_secrets_store` that has a minimum size 1 instead of 0?

Reply:
If we wanted to do that we would have to make `MessageSecretsStore` part of `MlsGroup` as well. So far it's only part of `ManagedGroup`. While not impossible, this would take away from the optional character it has now and it would make it harder to put it behind a feature flag. 
I'm also not convinced it would make things that much better, because the same branching logic would still have to exist – just elsewhere.
If we really wanted to do this, I'd prefer to do it in a follow-up PR because it would lead to a lot of code change.

Reply:
Ok, then please file a follow up.

Reply:
I filed #635.